dotnet/src/SemanticKernel.Core/PromptTemplate/NoopPromptTemplate.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// No operation implementation of <see cref="IPromptTemplate"/> that just returns the prompt template.
+/// </summary>
+internal sealed class NoopPromptTemplate : IPromptTemplate
+{
+    private readonly PromptTemplateConfig _promptConfig;
+
+    /// <summary>
+    /// Constructor for NoopPromptTemplate.
+    /// </summary>
+    /// <param name="promptConfig">Prompt template configuration</param>
+    internal NoopPromptTemplate(PromptTemplateConfig promptConfig)
+    {
+        Verify.NotNull(promptConfig, nameof(promptConfig));
+        Verify.NotNull(promptConfig.Template, nameof(promptConfig.Template));
+
+        this._promptConfig = promptConfig;
+    }
+
+    /// <inheritdoc/>
+    public Task<string> RenderAsync(Kernel kernel, KernelArguments? arguments = null, CancellationToken cancellationToken = default)
+    {
+        return Task.FromResult(this._promptConfig.Template);
```

Comment:
This task could be cached by the ctor instead of storing the the config, and then the same task returned every time.

Reply:
Doing that results in this warning https://github.com/Microsoft/vs-threading/blob/main/doc/analyzers/VSTHRD003.md and also will return the wring value if someone changes the `PromptTemplateConfig.Template`

Reply:
do that without usage of `await` and you may not have this warning.

Something like:

```csharp
   ctor() {
       this._renderResult = Task.FromResult(this._promptConfig.Template);
   }
   
   RenderAsync() => this._renderResult;
```

Reply:
Right, you wouldn't await the task, you'd just return it directly. That analyzer is also not really relevant to this code base in general.

As for the config changing, is that a supported pattern? Doesn't that already not work for all the other templates that parse the string in the ctor to make rendering faster?

Reply:
I'm not using `await`, I'll update and suppress the warning